python/pyspark/sql/tests.py

```diff
@@ -1435,6 +1435,12 @@ def test_time_with_timezone(self):
         self.assertEqual(now, now1)
         self.assertEqual(now, utcnow1)
 
+    # regression test for SPARK-19561
+    def test_datetime_at_epoch(self):
+        epoch = datetime.datetime.fromtimestamp(0)
+        df = self.spark.createDataFrame([Row(date=epoch)])
+        self.assertEqual(df.first()['date'], epoch)
```

Comment:
Yes, before this patch, `df.first()` is `Row(None)`.

I tried putting it in `DataTypeTests` first, but it was difficult to get a reasonable failing test case there. Python `int`s are up to 2^63 on 64-bit systems, so it doesn't overflow to long there. The issue is b/c Scala `int` are 32-bit, so Py4J is the part that converts it to long.

We could put the test there, but it doesn't really capture the issue IMO.